### PR TITLE
Ordered log lines

### PR DESCRIPTION
**Pull request by eranra**

### CODE REVIEW

**Comment by eranra** [comment]
Now logs look like::

image

File: log/logger.go [hunk 1]
```diff
@@ -14,13 +16,26 @@ import (
...
 
+// Line orders fields for json representation
+type Line struct {
+	Timestamp string                 `json:"time"`
+	File      string                 `json:"file"`
+	Line      string                 `json:"line"`
+	Verbosity string                 `json:"level"`
+	Component string                 `json:"component"`
+	Message   string                 `json:"message"`
+	Context   map[string]interface{} `json:"context"`
```
Comment (jcantrill) [inline log/logger.go:36]: Per your comment additional key/value are now placed under context which I believe changes the experience.  Not as pretty but maybe the "standard" keys should include underscores to distinguish them and we remove "context" :man_shrugging:

**Review by alanconway — CHANGES_REQUESTED** [review]
Ignore remarks about pull #15, got the two confused.

File: log/logger.go (hunk 1 above)
Comment (eranra) [reply log/logger.go:36]: @alanconway @jcantrill I need good advice (golang) on how to move the Context from being another field in the struct to set of keys and values that json will render ... but the main idea and reason to move to struct was to keep the order ... and this forced me to move the context to be field ...

Comment (jcantrill) [reply log/logger.go:36]: Can't you keep "context" but place it inline to accompish the same result? I'm not certain about order but as i commented elsewhere I would remove context and possibly prefix "lib" keys with underscores as anything in context is unlikely to collide

Comment (alanconway) [reply log/logger.go:36]: Here's a different solution - leave Line as a map[string]interface{} but give it a custom MarshalJSON function to do the sorting. Here's the code, it is based on the standard json.Marshal: 

```
package log

import (
	"bytes"
	"encoding/json"
	"reflect"
	"sort"
)

// rank gives the rank of built-in keys, other keys have rank 0.
// Log lines are encoded highest rank first.
var rank = map[string]int{
	"time":      5,
	"component": 4,
	"level":     3,
	"source":    2,
	"message":   1,
}

type Line map[string]interface{}

// MarshalJSON marshals Line as a JSON map with special sorting.
// Built-in keys go first, according to their rank.
// Other keys are sorted alphabetecally.
func (line Line) MarshalJSON() ([]byte, error) {
	if line == nil {
		return json.Marshal(line) // "null"
	}
	b := &bytes.Buffer{}

	// Extract keys and sort by rank.
	keys := make([]string, 0, len(line))
	for k := range line {
		keys = append(keys, k)
	}
	sort.Slice(keys, func(i, j int) bool {
		a, b := keys[i], keys[j]
		ra, rb := rank[a], rank[b]
		if ra == rb { // same rank, alphabetical sort
			return a < b
		} else {
			return ra > rb
		}
	})

	e := json.NewEncoder(b)
	b.WriteByte('{')
	for i, k := range keys {
		if i > 0 {
			b.WriteByte(',')
		}
		if err := e.Encode(k); err != nil {
			return nil, &json.MarshalerError{Type: reflect.TypeOf(line), Err: err}
		}
		b.WriteByte(':')
		if err := e.Encode(line[k]); err != nil {
			return nil, &json.MarshalerError{Type: reflect.TypeOf(line), Err: err}
		}
	}
	b.WriteByte('}')
	return b.Bytes(), nil
}

```

Comment (eranra) [reply log/logger.go]: @alanconway I incorporated all your remarks from PR #15 and closed the PR (in particular the file:line is now under developer flag)
@alanconway I decided to keep the line as a struct and only custom json work against the context field, this way things are more clean

Comment (eranra) [reply log/logger.go]: @jcantrill I added underscores as suggested

Comment (eranra) [reply log/logger.go]: Current version logs for developers looks like this::

image

**Review by alanconway — APPROVED** [review]
/lgtm

**Pull request merged**